packages/account-sdk/src/interface/public-utilities/spend-permission/methods/getHash.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  spendPermissionManagerAbi,
+  spendPermissionManagerAddress,
+} from ':sign/base-account/utils/constants.js';
+import { getClient } from ':store/chain-clients/utils.js';
+import { readContract } from 'viem/actions';
+import { SpendPermissionTypedData } from '../utils.js';
+
+/**
+ * Gets the hash of a spend permission from the SpendPermissionManager contract.
+ *
+ * This function calls the getHash method on the SpendPermissionManager contract
+ * to compute the unique hash for a given spend permission. This hash can be used
+ * to identify the permission on-chain and is required for operations like approval,
+ * revocation, and spending.
+ *
+ * @param permission - The spend permission message object from the typed data that
+ * contains all permission details.
+ * @param chainId - The chain ID to use for the contract call
+ *
+ * @returns A promise that resolves to the permission hash as a hex string
+ *
+ * @example
+ * ```typescript
+ * import { getHash } from '@base-org/account/spend-permission';
+ *
+ * const permission = {
+ *   account: '0x1234...',
+ *   spender: '0x5678...',
+ *   token: '0xabcd...',
+ *   allowance: '1000000', // 1 USDC (6 decimals)
+ *   period: 86400 * 30, // 30 days in seconds
+ *   start: Date.now(),
+ *   end: Date.now() + (86400 * 365),
+ *   salt: '0x1234567890abcdef...',
+ *   extraData: '0x'
+ * };
+ *
+ * const hash = await getHash(permission, 8453); // Base mainnet
+ * console.log('Permission hash:', hash);
+ * ```
+ */
+export const getHash = async (
```

Comment:
would be nice if we did this offline without calling the contract

Reply:
hmm leaning towards calling contract tho
* still thinks SDK should be as lean as possible
* we can honor contract as the single source of truth
* this call happens the same time as user approving spend permission and in the background, so no performance issue

Reply:
alright

btw are we sure this doesn't cause the popup to be blocked?

Reply:
am i understanding correctly that the SDK has to be in the connected state to get the hash?

Reply:
> are we sure this doesn't cause the popup to be blocked?

ahhh i haven't tested yet, but 
* we use `promise.all` to fire both at the same time, i think we should be good?
* also with the new popup blocked dialog, the UX for it wouldn't be too bad i think

Reply:
>  SDK has to be in the connected state to get the hash?

ya this is true, cause we only get the RPC url from wallet connection. But i think if dapp doesn't connect beforehand, they won't be having the spend permission hash to begin with

Reply:
not ideal if someone wants to calculate the spend permission hash server side

Reply:
but the hash has limited use anyway so nbd